src/test/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/resource/ReportResourceTest.kt

```diff
@@ -1444,6 +1444,377 @@ class ReportResourceTest : SqsIntegrationTestBase() {
+          .header("Content-Type", "application/json")
+          .bodyValue(validPayload)
+          .exchange()
+          .expectStatus().isNotFound
+
+        assertThatNoDomainEventsWereSent()
+      }
+
+      @Test
+      fun `cannot change type of a report with invalid payload`() {
+        webTestClient.patch().uri(urlWithNoQuestionsOrHistory)
+          .headers(setAuthorisation(roles = listOf("ROLE_MAINTAIN_INCIDENT_REPORTS"), scopes = listOf("write")))
+          .header("Content-Type", "application/json")
+          .bodyValue(
+            // language=json
+            """
+              {"type": "DAMAGE"}
+            """,
+          )
+          .exchange()
+          .expectStatus().isBadRequest
+
+        assertThatNoDomainEventsWereSent()
+      }
+
+      @Test
+      fun `cannot change type of a report to an unknown one`() {
+        webTestClient.patch().uri(urlWithNoQuestionsOrHistory)
+          .headers(setAuthorisation(roles = listOf("ROLE_MAINTAIN_INCIDENT_REPORTS"), scopes = listOf("write")))
+          .header("Content-Type", "application/json")
+          .bodyValue(
+            // language=json
+            """
+              {"newType": "MISSING_TYPE"}
+            """,
+          )
+          .exchange()
+          .expectStatus().isBadRequest
+
+        assertThatNoDomainEventsWereSent()
+      }
+
+      @Test
+      fun `cannot change type of a report to inactive one`() {
+        webTestClient.patch().uri(urlWithNoQuestionsOrHistory)
+          .headers(setAuthorisation(roles = listOf("ROLE_MAINTAIN_INCIDENT_REPORTS"), scopes = listOf("write")))
+          .header("Content-Type", "application/json")
+          .bodyValue(
+            // language=json
+            """
+              {"newType": "OLD_ASSAULT3"}
+            """,
+          )
+          .exchange()
+          .expectStatus().isBadRequest
+
+        assertThatNoDomainEventsWereSent()
+      }
+    }
+
+    @DisplayName("works")
+    @Nested
+    inner class HappyPath {
+      @Test
+      fun `can change type of a report but keeping it the same`() {
+        val reportWithQuestions = reportRepository.save(
+          buildIncidentReport(
+            incidentNumber = "IR-0000000001124146",
+            reportTime = now.minusMinutes(3),
+            status = Status.AWAITING_ANALYSIS,
+            generateQuestions = 2,
+            generateResponses = 2,
+            generateHistory = 0,
+          ),
+        )
+        webTestClient.patch().uri("/incident-reports/${reportWithQuestions.id}/type")
+          .headers(setAuthorisation(roles = listOf("ROLE_MAINTAIN_INCIDENT_REPORTS"), scopes = listOf("write")))
+          .header("Content-Type", "application/json")
+          .bodyValue(
+            // language=json
+            """
+              {"newType": "FINDS"}
+            """,
+          )
+          .exchange()
+          .expectStatus().isOk
```

Comment:
Should this be allowed given type is not changing? 🤔

Reply:
i think it should 404 if not found, but it's better to be idempotent in REST apis where possible. so if you set type to DAMAGE twice the second time should be ok but make no changes